tests/run.sh

```diff
@@ -105,14 +103,23 @@ flowctl-go api build --directory ${TESTDIR}/builds --build-id test-build-id --so
 
 # Activate the catalog.
 flowctl-go api activate --build-id test-build-id --all --network "flow-test" --log.level info || bail "Activate failed."
-# Wait for a data-flow pass to finish.
-flowctl-go api await --build-id test-build-id --log.level info || bail "Await failed."
-# Read out materialization results.
-sqlite3 -header "${OUTPUT_DB}" "select id, canary from test_results;" > "${ACTUAL}"
+
```

Comment:
Continue to use `api await` here? Seems that the capture connectors in question _do_ actually exit, given the source-test modification, and we can continue to use this property.

Think this makes test UX better since `api await` will fail-fast if a shard has failed, and exits if the dataflow has settled even if the output < expected lines (rather than hanging).

Reply:
From my tests, they don't exit. In particular source-postgres and source-mysql will wait until they reach a nonexistant-watermark and will never quit. See this run for example: https://github.com/estuary/connectors/actions/runs/3553412074/jobs/5968911590#step:14:204

Reply:
But we do probably need some sort of timeout here in case we don't reach the expected number of lines

Reply:
🤦 of course. Sounds good